docs/Fleet management Design.md

```diff
@@ -143,16 +143,41 @@ perform in the member cluster joining process
 
 1. The controller "claims" the member cluster by creating a "lease" configMap in the member cluster.
    * The configMap's name is a well known name by convention.
-   * It contains the fleet UID which can be reused if the hub is migrated to another cluster.
+   * It contains the fleet UID so that one cluster cannot join multiple fleet. We can use the UID of the hub cluster as the fleet UID for now.
    * The controller will reject the member cluster if the configMap already contains a fleet ID not the same as its ID.
-   * We could use a CR but that means we also need to apply a CRD there.
    * We can add more information to the configMap such as the join time, last health check result, last health check time, etc.
 2. The controller creates a namespace of the same name as the member cluster in the hub cluster.
 3. The controller starts a health check loop for each member cluster which
    * Checks the health of the member cluster by updating the last health check time in the lease configMap
    * Lists all the nodes in a member cluster and aggregates their `Capacity` and `Allocatable`
    * Updating the `managedCluster` condition status
 
+#### member cluster lease
+Since a configMAP has no schema, an alternative way is to create a CRD as the lease so that we can have a more defined
```

Comment:
The key value pair of the ConfigMap type would allow for any schema of the value. A serialized Lease object would work as the value, and the key could simply be the FQDN of the Lease CRD (which exists on the Hub).

Reply:
yeah,  if we go down the CRD route then we probably don't need a configMap anymore.